Library/Homebrew/cmd/test-bot.rb

```diff
@@ -590,6 +590,11 @@ def formula(formula_name)
           if bottle_step.passed? && bottle_step.has_output?
             bottle_filename =
               bottle_step.output.gsub(/.*(\.\/\S+#{bottle_native_regex}).*/m, '\1')
+            bottle_rb_filename = bottle_filename.gsub(/\.(\d+\.)?tar\.gz$/, ".rb")
+            bottle_merge_args = ["--merge", "--write", "--no-commit", bottle_rb_filename]
```

Comment:
I'm not talking avoid re-download but avoid rebuilding from source. From what I'm seeing, some formula will be built twice or more when we process PR that bumps revisions. The root cause is [we uninstall the formula after test it](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/cmd/test-bot.rb#L621). So if another formula depends on the previous formula, it will be rebuilt.